unified-runtime/scripts/templates/helper.py

```diff
@@ -127,264 +114,261 @@ class type_traits:
     RE_ARRAY = r"(.*)\[([1-9][0-9]*)\]"
 
     @staticmethod
-    def base(name):
-        return _remove_const_ptr(name)
+    def base(name) -> Union[str, bool]:
+        try:
+            return _remove_const_ptr(name)
+        except BaseException:
+            return False
 
     @classmethod
-    def is_handle(cls, name):
+    def is_handle(cls, name) -> bool:
         try:
-            return True if re.match(cls.RE_HANDLE, name) else False
-        except:
+            return bool(re.match(cls.RE_HANDLE, name))
+        except TypeError:
             return False
 
     @classmethod
-    def is_native_handle(cls, name):
+    def is_native_handle(cls, name) -> bool:
         try:
-            return True if re.match(cls.RE_NATIVE_HANDLE, name) else False
-        except:
+            return bool(re.match(cls.RE_NATIVE_HANDLE, name))
+        except TypeError:
             return False
 
     @classmethod
-    def is_pointer_to_pointer(cls, name):
+    def is_pointer_to_pointer(cls, name) -> bool:
         try:
-            return True if re.match(cls.RE_PPOINTER, name) else False
-        except:
+            return bool(re.match(cls.RE_PPOINTER, name))
+        except TypeError:
             return False
 
     @classmethod
-    def is_ipc_handle(cls, name):
+    def is_ipc_handle(cls, name) -> bool:
         try:
-            return True if re.match(cls.RE_IPC, name) else False
-        except:
+            return bool(re.match(cls.RE_IPC, name))
+        except TypeError:
             return False
 
     @staticmethod
-    def is_class_handle(name, meta):
+    def is_class_handle(name, meta) -> bool:
         try:
             name = _remove_const_ptr(name)
             return len(meta["handle"][name]["class"]) > 0
-        except:
+        except BaseException:
             return False
 
     @classmethod
-    def is_pointer(cls, name):
+    def is_pointer(cls, name) -> bool:
         try:
-            return True if re.match(cls.RE_POINTER, name) else False
-        except:
+            return bool(re.match(cls.RE_POINTER, name))
```

Comment:
Would `re.match(...) is not None` be more idiomatic?

Reply:
It's certainly more typing. `None`ness is equivalent to truthiness when uses in `if` statements. My linter (pyright) doesn't warn about conversion to bool like this, it does warn about using `re.match(cls.RE_POINTER, name) == None` instead of `re.match(cls.RE_POINTER, name) is None`.